frontend/src/metabase/common/hooks/use-docs-url/use-docs-url.ts

```diff
@@ -1,10 +1,27 @@
 import { useSelector } from "metabase/lib/redux";
-import { getDocsUrl } from "metabase/selectors/settings";
+import { type UtmProps, getDocsUrl } from "metabase/selectors/settings";
 import { getShowMetabaseLinks } from "metabase/selectors/whitelabel";
 
-export const useDocsUrl = (page: string, anchor?: string) => {
+type DocsUtmProps = Omit<UtmProps, "utm_medium"> & {
+  utm_medium?: UtmProps["utm_medium"];
+};
+
+export const useDocsUrl = (
+  page: string,
+  anchor?: string,
+  utm?: DocsUtmProps,
```

Comment:
I agree, but wanted to avoid the large amount of changes that would create in calling code.

I could also add an overload, so the old invocation would keep working, or do we want to avoid that too?

Reply:
> I agree, but wanted to avoid the large amount of changes that would create in calling code.

We can do this in a follow-up PR if diff would be too large

> I could also add an overload, so the old invocation would keep working, or do we want to avoid that too?

My vote is to keep things simple :)

Reply:
Did just that now!